### PR TITLE
Onboarding Intro - Add choose address bar position

### PR DESCRIPTION
**Pull request by alessandroboron**
Task/Issue URL: https://app.asana.com/0/1206329551987282/1208084960726997/f

**Description**:

Add Address Bar position selection screen to the onboarding.
[FIGMA DESIGN](https://www.figma.com/design/s9cFusskywDt9NFNFuqRcG/Onboarding---Highlights?node-id=7419-88693&m=dev)

Steps to test this PR:
PREREQUISITES: login as an internal user at use-login.duckduckgo.com

**Steps to test this PR**:

**SCENARIO 1 - Normal Flow **

1. Tap the ⚙️ icon located on the Address bar right-hand side. (iPad tap … button first)
2. Scroll to the bottom and select All Debug Options under Debug section.
3. Scroll to the bottom and select New Onboarding to navigate to the Onboarding debug section.
4. Ensure that Onboarding Highlights local settings enabled is turned off.
5. Tap on Preview New Onboarding Intro.
6. Wait for the onboarding intro to appear.
7. Tap the “Let’s do it Button”.
8. The browser comparison chart should appear.
9. Ensure no progress bar is shown.
10. Tap the “Skip” button.
**Expected Result:** The onboarding should dismiss.

**SCENARIO 2 - Highlights Flow - iPhone**

1. Tap the ⚙️ icon located on the Address bar right-hand side. (iPad tap … button first)
2. Scroll to the bottom and select All Debug Options under Debug section.
3. Scroll to the bottom and select New Onboarding to navigate to the Onboarding debug section.
4. Ensure that Onboarding Highlights local settings enabled is turned on.
5. Tap on Preview New Onboarding Intro.
6. Wait for the onboarding intro to appear.
7. Tap the “Let’s do it Button”.
8. The browser comparison chart should appear.
9. Tap the “Skip” button.
10. Wait for the chose app icon dialog to appear.
11. Tap the “Next” button.
17. Wait for the Address Bar position dialog to apper.
18. Ensure that the current browser address bar position is selected.
19. Select the other address bar position.
16. Tap the “Next” button.
15. The onboarding should dismiss.
17. Exit the Debug and Settings screen.
**Expected Result:** The Address bar should be in the position previously selected during the onboarding.

**SCENARIO 2 - Highlights Flow - iPad**

1. Tap the ⚙️ icon located on the Address bar right-hand side. (iPad tap … button first)
2. Scroll to the bottom and select All Debug Options under Debug section.
3. Scroll to the bottom and select New Onboarding to navigate to the Onboarding debug section.
4. Ensure that Onboarding Highlights local settings enabled is turned on.
5. Tap on Preview New Onboarding Intro.
6. Wait for the onboarding intro to appear.
7. Tap the “Let’s do it Button”.
8. The browser comparison chart should appear.
9. Tap the “Skip” button.
10. Wait for the chose app icon dialog to appear.
11. Tap the “Next” button.
**Expected Result:** The onboarding should dismiss.

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/DaxOnboarding.xcassets/address-bar-bottom.imageset/address-bar-bottom.pdf
Comment (SabrinaTardio) [inline DuckDuckGo/DaxOnboarding.xcassets/address-bar-bottom.imageset/address-bar-bottom.pdf:1]: I think we normally use SVG?

File: Core/NSAttributedStringExtension.swift [hunk 1]
```diff
@@ -40,4 +40,86 @@ extension NSAttributedString {
         newString.setAttributes([.font: newFont], range: string.fullRange)
         return newString
     }
+
+    /// Creates a new `NSAttributedString` initialized with the characters and attributes of the current attributed string plus the specified font.
```
Comment (SabrinaTardio) [inline Core/NSAttributedStringExtension.swift:44]: Nice!

File: DuckDuckGoTests/OnboardingAddressBarPositionPickerViewModelTests.swift [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    func testWhenUpdateAddressBarThenDisplayModelsAreUpdated() throws {
+        // GIVEN
+        addressBarPositionManagerMock.currentAddressBarPosition = .top
+        let sut = OnboardingAddressBarPositionPickerViewModel(addressBarPositionManager: addressBarPositionManagerMock)
+        XCTAssertEqual(sut.items.first?.type, .top)
+        XCTAssertTrue(sut.items.first?.isSelected ?? false)
+
+        // WHEN
+        sut.setAddressBar(position: .bottom)
```
Comment (SabrinaTardio) [inline DuckDuckGoTests/OnboardingAddressBarPositionPickerViewModelTests.swift:68]: Shouldn’t we also test that addressBarPositionManager.currentAddressBarPosition has been updated?

**Review by SabrinaTardio — APPROVED** [review]
LGTM!

left a couple of minor comments

File: DuckDuckGo/DaxOnboarding.xcassets/address-bar-bottom.imageset/address-bar-bottom.pdf
Comment (alessandroboron) [reply DuckDuckGo/DaxOnboarding.xcassets/address-bar-bottom.imageset/address-bar-bottom.pdf:1]: Most of our assets are actually .pdf files.

File: DuckDuckGoTests/OnboardingAddressBarPositionPickerViewModelTests.swift (hunk 2 above)
Comment (alessandroboron) [reply DuckDuckGoTests/OnboardingAddressBarPositionPickerViewModelTests.swift:68]: good catch, updated!

**Pull request merged**